homeassistant/components/camera/yi.py

```diff
@@ -57,68 +55,67 @@ def __init__(self, hass, config):
         self._last_image = None
         self._last_url = None
         self._manager = hass.data[DATA_FFMPEG]
-        self._name = config.get(CONF_NAME)
-        self.host = config.get(CONF_HOST)
-        self.port = config.get(CONF_PORT)
-        self.path = config.get(CONF_PATH)
-        self.user = config.get(CONF_USERNAME)
-        self.passwd = config.get(CONF_PASSWORD)
-
-    @property
-    def name(self):
-        """Return the name of this camera."""
-        return self._name
+        self._name = config[CONF_NAME]
+        self.host = config[CONF_HOST]
+        self.port = config[CONF_PORT]
+        self.path = config[CONF_PATH]
+        self.user = config[CONF_USERNAME]
+        self.passwd = config[CONF_PASSWORD]
 
     @property
     def brand(self):
         """Camera brand."""
         return DEFAULT_BRAND
 
-    def get_latest_video_url(self):
-        """Retrieve the latest video file from the customized Yi FTP server."""
-        from ftplib import FTP, error_perm
+    @property
+    def name(self):
+        """Return the name of this camera."""
+        return self._name
 
-        ftp = FTP(self.host)
-        try:
-            ftp.login(self.user, self.passwd)
-        except error_perm as exc:
-            _LOGGER.error('There was an error while logging into the camera')
-            _LOGGER.debug(exc)
-            return False
+    async def _get_latest_video_url(self):
+        """Retrieve the latest video file from the customized Yi FTP server."""
+        from aioftp import Client, StatusCodeError
 
+        ftp = Client()
         try:
-            ftp.cwd(self.path)
-        except error_perm as exc:
-            _LOGGER.error('Unable to find path: %s', self.path)
-            _LOGGER.debug(exc)
-            return False
-
-        dirs = [d for d in ftp.nlst() if '.' not in d]
-        if not dirs:
-            _LOGGER.warning("There don't appear to be any uploaded videos")
-            return False
-
-        latest_dir = dirs[-1]
-        ftp.cwd(latest_dir)
-        videos = ftp.nlst()
-        if not videos:
-            _LOGGER.info('Video folder "%s" is empty; delaying', latest_dir)
-            return False
-
-        return 'ftp://{0}:{1}@{2}:{3}{4}/{5}/{6}'.format(
-            self.user, self.passwd, self.host, self.port, self.path,
-            latest_dir, videos[-1])
+            await ftp.connect(self.host)
```

Comment:
Maybe For a future PR if it's possible: Opening a whole new FTP session with a new socket for each new image is ... a *bit* inefficient I think.

Reply:
Great feedback; I think I should be able to address this easily. Hang tight.

Reply:
Ah just realized that ffmpeg/whatever is passing on the video stream is opening a new socket for each image anyway... So this is probably not that big of a speedup in the end. Sorry about that!